lib/ash/info.ex

```diff
@@ -49,4 +49,32 @@ defmodule Ash.Info do
         )
     end
   end
+
+  @doc """
+  Returns a list of extensions in use by all the domains and resources in the
+  given application.
+  """
+  @spec extensions_in_use(app :: Application.app()) :: [module()]
+  def extensions_in_use(app) do
+    app
+    |> domains()
+    |> Enum.flat_map(&Ash.Domain.Info.extensions(&1, include_resource_extensions?: true))
+    |> Enum.uniq()
+  end
+
+  @doc """
+  Returns a list of all defined extensions in the application.
+
+  > #### Speed {: .warning}
+  >
+  > This function will load all modules in the application, which can be slow.
+  > Do not call this function in time sensitive code paths and prefer calling it
+  > at compile time if possible.
```

Comment:
at compile time I think it will actually have weird behavior. Moreso like "this is primarily for use in mix tasks"